src/Altinn.App.Api/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -104,6 +109,40 @@ private static void AddApplicationInsights(IServiceCollection services, IConfigu
             }
         }
 
+        private static void AddOpenTelemetry(IServiceCollection services, IConfiguration config, IWebHostEnvironment env)
+        {
+            services
+                .AddOpenTelemetry()
+                .ConfigureResource(r => 
+                    r.AddService(
+                        serviceName: config.GetValue<string>("ServiceName") ?? "unknown",
+                        serviceVersion: typeof(Program).Assembly.GetName().Version?.ToString(),
+                        serviceInstanceId: Environment.MachineName))
+                .WithTracing(tpbuilder => 
+                {
+                    if (env.IsDevelopment())
+                    {
+                        tpbuilder.SetSampler(new AlwaysOnSampler());
+                    }
+
+                    tpbuilder
+                        .AddSource("TODO: find docker-image-tag/assembly-version")
+                        .AddHttpClientInstrumentation()
```

Comment:
We should go through the options for the instrumentations as well, most of them can be configured to include exceptions as span events, for example:
```csharp
.AddHttpClientInstrumentation(opts => opts.RecordException = true)
```